### PR TITLE
Et 4967 tika

### PR DESCRIPTION
**Pull request by joecks**
Adds the new tika service

### CODE REVIEW

File: b2b/service/documents/main.tf [hunk 1]
```diff
@@ -36,18 +37,18 @@ data "aws_iam_policy_document" "ecs_service_role" {
   }
 }
 
-resource "aws_iam_role" "sagemaker" {
+resource "aws_iam_role" "task_role" {
   count = local.create_task_role ? 1 : 0
 
   name               = "${title(var.tenant)}DocumentsEcsTaskRole"
-  description        = "Allows the ECS Task to access sagemaker"
+  description        = "A task role for outgoing request for the Document Service: Like to Tika or Sagemaker."
```
Comment (Robert-Steiner) [inline b2b/service/documents/main.tf:44]: I can't find the `aws_iam_policy_document` with the permissions that are required to communicate to the tika service. I wonder why we would such a policy. service to service communication should work without it or do i missing something?

File: b2b/service/documents/main.tf [hunk 2]
```diff
@@ -170,16 +171,19 @@ module "service" {
...
     XAYN_WEB_API__EMBEDDING__TOKEN_SIZE = var.token_size
     },
-    local.create_task_role && try(var.sagemaker_endpoint.target_model, null) != null ? { XAYN_WEB_API__EMBEDDING__TARGET_MODEL = var.sagemaker_endpoint.target_model } : {},
-    local.create_task_role && try(var.sagemaker_endpoint.max_retries, null) != null ? { XAYN_WEB_API__EMBEDDING__RETRY_MAX_ATTEMPTS = var.sagemaker_endpoint.max_retries } : {}
+    local.sagemaker_enabled && try(var.sagemaker_endpoint.target_model, null) != null ? { XAYN_WEB_API__EMBEDDING__TARGET_MODEL = var.sagemaker_endpoint.target_model } : {},
+    local.sagemaker_enabled && try(var.sagemaker_endpoint.max_retries, null) != null ? { XAYN_WEB_API__EMBEDDING__RETRY_MAX_ATTEMPTS = var.sagemaker_endpoint.max_retries } : {},
+    var.tika_configuration.enabled ? { XAYN_WEB_API__TEXT_EXTRACTOR__ENABLED = var.tika_configuration.enabled } : {},
+    var.tika_configuration.enabled ? { XAYN_WEB_API__TEXT_EXTRACTOR__URL = var.tika_configuration.endpoint } : {},
+    var.tika_configuration.enabled ? { XAYN_WEB_API__TEXT_EXTRACTOR__ALLOWED_CONTENT_TYPE = jsonencode(var.tika_configuration.allowed_conentent_type) } : {}
```
Comment (Robert-Steiner) [inline b2b/service/documents/main.tf:186]:
```suggestion
    var.tika_configuration.enabled ? { 
    XAYN_WEB_API__TEXT_EXTRACTOR__ENABLED = var.tika_configuration.enabled, 
    XAYN_WEB_API__TEXT_EXTRACTOR__URL = var.tika_configuration.endpoint,
    XAYN_WEB_API__TEXT_EXTRACTOR__ALLOWED_CONTENT_TYPE = jsonencode(var.tika_configuration.allowed_conentent_type) 
    } : {},

```

File: b2b/service/tika/main.tf [hunk 3]
```diff
@@ -0,0 +1,110 @@
...
+
+  alb = {
+    listener_arn             = var.alb_listener_arn
+    listener_port            = var.alb_listener_port
+    health_path              = "/"
+    routing_header_condition = null
+
+    rules = [
+      {
+        routing_path_pattern = ["/rmeta", "/rmeta/*"]
```
Comment (Robert-Steiner) [inline b2b/service/tika/main.tf:62]: nice that we can reuse the current alb 👍

File: generic/service/service/main.tf [hunk 4]
```diff
@@ -126,6 +126,7 @@ resource "aws_lb_target_group" "service" {
   vpc_id      = var.vpc_id
   target_type = "ip"
 
+
```
Comment (Robert-Steiner) [inline generic/service/service/main.tf:129]:
```suggestion
```

File: b2b/service/documents/main.tf (hunk 2 above)
Comment (joecks) [reply b2b/service/documents/main.tf:186]: somehow, I couldn't do that. The reason is pretty strange:

type wise this seems to be true: 
{ }  === { x=a }
{ }  !== { x=a,  y=b }

So if I remove at least two entries the linter and interpreter would complain.

Comment (Robert-Steiner) [reply b2b/service/documents/main.tf:186]: could be that all values need to have the same type. we can leave it as it is

**Pull request merged**